api/overview/paging.adoc

```diff
@@ -0,0 +1,79 @@
+
+[[paging]]
+== Paging
+
+Paging is enabled by default. When querying collections and the _limit_ query parameter is not specified, only 1000
+resources are returned by default. This is driven by the new advanced setting:
+
+[source,data]
+----
+:api:
+  :max_results_per_page: 1000
+----
+
+Example GET with offset and limit specified.
+
+[source,data]
+----
+GET /api/vms?offset=2&limit=2&sort_by=name&expand=resources
+----
+
+[source,json]
+----
+{
+  "name"      : "vms",
+  "count"     : 7,
+  "subcount"  : 2,
+  "pages"     : 4,
+  "resources" : [
+    {
+      "href"   : "http://localhost:3000/api/vms/7254",
+      "id"     :  "7254",
+      "vendor" : "vmware",
+      "name"   : "test-vm1",
+      ...
+    },
+    {
+      "href"   : "http://localhost:3000/api/vms/7257",
+      "id"     : "7257",
+      "vendor" : "vmware",
+      "name"   : "test-vm2",
+      ...
+    }
+  ],
+ "actions" : [
+    ...
+  ],
+ "links" : {
+   "self"     : "http://localhost:3000/api/vms?offset=2&limit=2&sort_by=name&expand=resources",
+   "next"     : "http://localhost:3000/api/vms?offset=4&limit=2&sort_by=name&expand=resources",
+   "previous" : "http://localhost:3000/api/vms?offset=0&limit=2&sort_by=name&expand=resources",
+   "first"    : "http://localhost:3000/api/vms?offset=0&limit=2&sort_by=name&expand=resources",
+   "last"     : "http://localhost:3000/api/vms?offset=6&limit=2&sort_by=name&expand=resources"
+ }
+}
+----
+
+With paging enabled by default, collection queries now return the following:
+
+* "pages" metadata which is the total number of pages for the _limit_ specified (or the default 1000).
+* "links" metadata section which includes the necessary URL's for fetching the current page,
+next and previous pages, as well as the first and last pages.
+
+_Notes:_
+
+* When the page being returned is the first page (i.e. offset = 0), the _previous_ link is not returned.
+* When the page being returned is the last page, the _next_ page link is not returned.
+* For small result sets where _limit_ is less than max_results_per_page, the _limit_ will be honored.
+* For large result sets where the _limit_ is greater than max_results_per_page, _limit_ will be ignored and
+the result will be limited to the first max_results_per_page resources.
+
+[NOTE]
+.About max_results_per_page
+====
+
+If the default value for max_results_per_page does not meet your needs, It can be changed using 
+the Advanced Settings feature in the UI. It is recommended however that it is not set to some arbitrarily large
+number in an effort to circumvent the paging feature - it is designed to prevent excessive load, and certainly
+requesting millions of records which is easy to do with the metric rollups/event streams APIs.
```

Comment:
This sentence feels incomplete. Perhaps something along the lines of:
"...designed to prevent excessive load, which is easy to do with collections such as metric rollups and event streams which may have millions of records"?